### ISSUE TITLE
Test using parallel async mocks hangs

### ISSUE THREAD

**Issue by divergentdave** [lipanski/mockito#166]
I have a test that uses multiple mocks in parallel async tasks, and it is newly hanging with mockito 1.0.0. The following minimized test reproduces the issue.

```
#[tokio::test]
async fn test_parallel_async() {
    futures::future::join_all((0..10).map(|_| async move {
        let mut s = Server::new_async().await;
        s.mock("POST", "/").create_async().await;

        reqwest::Client::new().post(s.url()).send().await.unwrap();
    }))
    .await;
}
```

Moreover, adding this to the test suite causes a number of other tests to hang. FWIW, I hooked up tokio-console to my test, and it warned that tasks had lost their wakers, but I'm not sure how trustworthy that diagnostic is.

**Comment by lipanski** [comment]
@divergentdave if you replace `Server::new_async()` with `Server::new_with_port_async(0)` it should work. this means the problem lies with the sever pool (semaphores?). I'll have a look when I get some time but bypassing the server pool should be a decent workaround, unless you intend to spawn thousands of servers for this test.

**Comment by lipanski** [comment]
@divergentdave this has been fixed in [1.0.1](https://github.com/lipanski/mockito/releases/tag/1.0.1). you can now go back to using `Server::new_async` and the server pool.

**Comment by divergentdave** [comment]
Thanks! The pool is working for me now.
